### PR TITLE
More review fixes: smart pointers, cleaner namespaces, naming fixes

### PR DESCRIPTION
**Pull request by ariostas**
I'll address some more review comments on this PR.

Changes made so far:
- Started moving to using smart pointers instead of raw pointers. (Changed the main ones, but I'll check if it's worth doing in other places)
-  Cleaned up namespaces (mainly removing a bunch of `::lst::`)

### CODE REVIEW

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc [hunk 1]
```diff
@@ -72,90 +72,71 @@ void Event::resetEventSync() {
     }
   }
   if (hitsInGPU) {
-    delete hitsInGPU;
-    delete hitsBuffers;
-    hitsInGPU = nullptr;
+    hitsInGPU = std::nullopt;
+    hitsBuffers = std::nullopt;
   }
```
Comment (fwyzard) [inline RecoTracker/LSTCore/src/alpaka/Event.dev.cc:77]: You can simply call `reset()` and let it do the check:
```c++
  hitsInGPU.reset();
  hitsBuffers.reset();
```

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc [hunk 2]
```diff
@@ -168,15 +149,15 @@ void Event::addHitToEvent(std::vector<float> const& x,
   unsigned int nHits = x.size();
 
   // Initialize space on device/host for next event.
-  if (hitsInGPU == nullptr) {
-    hitsInGPU = new Hits();
-    hitsBuffers = new HitsBuffer<Device>(nModules_, nHits, devAcc, queue);
+  if (!hitsInGPU) {
+    hitsInGPU = Hits();
+    hitsBuffers = HitsBuffer<Device>(nModules_, nHits, devAcc, queue);
```
Comment (fwyzard) [inline RecoTracker/LSTCore/src/alpaka/Event.dev.cc:154]: I think
```suggestion
    hitsBuffers.emplace(nModules_, nHits, devAcc, queue);
```
would avoid the copy.

**Comment by ariostas** [comment]
I finished addressing the last batch of simple fixes suggested by Andrea:
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1724885193
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1724885712
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1724886322
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1724888548

As well as his other suggestions in this PR.

I force-pushed to squash some of the commits since we're starting to get close to the maximum number of commits for a CMSSW PR. (Although we'll probably have to squash more things later regardless)

**Review by slava77 — APPROVED** [review]

**Comment by slava77** [comment]
while rebasing I noticed that a few `::lst::PixelType` are left over after this PR, I did a git grep `::lst`:
- `lst.cc:  std::vector<std::vector<::lst::PixelType>> out_pixelType_vec;`
- `trkCore.cc:  std::vector<::lst::PixelType> pixelType_vec;` (and a bunch more lines)

**Comment by ariostas** [comment]
> > while rebasing I noticed that a few ::lst::PixelType are left over after this PR
> 
> Oops, sorry about that. I'll fix those this week.

I added a commit to #88 fixing the lines I missed. It was a very minor change, so I didn't think it needed its own PR.

**Pull request merged**